examples/server/server.h

```diff
@@ -26,6 +26,7 @@
 
 class Server {
 public:
+  virtual ~Server() = default;
```

Comment:
Same here, this isn't necessary.

Reply:
Since this is a kind of a teaching media, I think that it should be as correct as possible. 

According to CPP Core Guidelines (https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rc-dtor-virtual) a destructor of a base class should be either non-`virtual` and `protected` OR `virtual` and `public`. Since, I guess making the destructor `protected` (and then guarantee what you want, i.e., there is no virtual destruction) would entice more questions than the "normal" implementation, I think doing the right thing here is better for your audience.